File: 219.contains-duplicate-ii.py
#
# @lc app=leetcode id=219 lang=python3
#
# [219] Contains Duplicate II
#

# Time:  O(n)
# Space: O(n)

from typing import List


# @lc code=start
class Solution:
    def containsNearbyDuplicate(self, nums: List[int], k: int) -> bool:
        lookup = {}
        for i, num in enumerate(nums):
            if num in lookup and i - lookup[num] <= k:
                return True
            lookup[num] = i
        return False


# @lc code=end

File: 219.contains-duplicate-ii.go
/*
 * @lc app=leetcode id=219 lang=golang
 *
 * [219] Contains Duplicate II
 */
package leetcode

// @lc code=start
func containsNearbyDuplicate(nums []int, k int) bool {
	lookup := map[int]int{}
	for i := 0; i < len(nums); i++ {
		if _, ok := lookup[nums[i]]; ok && i-lookup[nums[i]] <= k {
			return true
		}
		lookup[nums[i]] = i
	}
	return false
}

// @lc code=end
